tests/tas-elasticsearch/src/test/java/org/alfresco/elasticsearch/ElasticsearchAdminConsoleTests.java

```diff
@@ -0,0 +1,151 @@
+    public DataSite dataSite;
+
+    @Autowired
+    protected ServerHealth serverHealth;
+
+    @Autowired
+    protected RestWrapper client;
+
+    private UserModel user;
+    private SiteModel siteModel;
+
+    /** Create a user and a private site and wait for these to be indexed. */
+    @BeforeClass (alwaysRun = true)
+    public void dataPreparation() throws Exception
+    {
+        serverHealth.assertServerIsOnline();
+
+        Step.STEP("Create a test user and private site.");
+        user = dataUser.createRandomTestUser();
+        siteModel = dataSite.usingUser(user).createPrivateRandomSite();
+        Step.STEP("Create a document in the site and wait for it to be indexed");
+        // We wait for this file to be indexed as a proxy for indexing probably being up to date.
+        createDocumentAndEnsureIndexed("DummyFile" + UUID.randomUUID() + ".txt");
+    }
+
+    /** Check the "elasticsearch" subsystem is selected and that the creation of a new document is reflected in the document counts. */
+    @TestRail (section = {
+            TestGroup.SEARCH }, executionType = ExecutionType.REGRESSION, description = "TODO.")
+    @Test (groups = { TestGroup.SEARCH })
+    public void elasticsearchAdminConsolePage() throws Exception
+    {
+        Step.STEP("Load the Search admin page.");
+        Document document = loadSearchAdminPage();
+
+        Step.STEP("Check that elasticsearch is selected and displayed.");
+        assertEquals(document.select("option[value=elasticsearch]").attr("selected"), "selected",
+                "Expected elasticsearch subsystem to be shown as selected.");
+        assertFalse(document.select("#elasticsearchSearch").hasClass("hidden"), "Expected elasticsearch section to be displayed.");
+
+        Step.STEP("Get the number of indexed and indexable documents.");
+        int indexedDocs = Integer.valueOf(document.select("#elasticsearchDocumentCount span.value").text());
+        int indexableDocs = Integer.valueOf(document.select("#repositoryNodesCount span.value").text());
+
+        Step.STEP("Create a document and wait for it to be indexed.");
+        createDocumentAndEnsureIndexed(FILE_NAME);
+
+        Step.STEP("Refresh the admin console.");
+        document = loadSearchAdminPage();
+        Step.STEP("Get the new number of indexed and indexable documents.");
+        int newIndexedDocs = Integer.valueOf(document.select("#elasticsearchDocumentCount span.value").text());
+        int newIndexableDocs = Integer.valueOf(document.select("#repositoryNodesCount span.value").text());
+
+        // Check the numbers have increased (in theory by one each, but there may be other documents being indexed too.
+        Assert.assertTrue(newIndexedDocs > indexedDocs, "Expected the number of indexed documents to increase, was "
+                + indexedDocs + ", now " + newIndexedDocs);
+        Assert.assertTrue(newIndexableDocs > indexableDocs, "Expected the number of indexable documents to increase, was "
```

Comment:
because we are indexing only one document, can we verify that `newIndexableDocs = indexableDocs+1`?

Reply:
I noticed the comment, why there are other documents being indexed too?

Reply:
There weren't any other documents being indexed in my local testing (and checking `y == x + 1` worked), but I wasn't sure that we could guarantee this would always be the case. I'm happy to change it for now, and we could change it back if we see intermittent failures?